model_interfaces/interface.py

```diff
@@ -70,7 +62,6 @@ def save(self):
     def load(self):
         pass
 
-
-class ChatSessionFactory(ABC):
-    def create_session(self) -> ChatSession:
-        pass
+    def token_len(self, text: str) -> int:
+        enc = tiktoken.get_encoding("cl100k_base")
+        return len(enc.encode(text))
```

Comment:
Why is this class taking a different route for computing the number of tokens?

Reply:
So this is here as the default tokeniser for a model, since not all models should supply their own. Since GPT is more common, I decided that should be the default. Though I could now just have this function call the `utils.llm` functions for tokenising with a GPT model, and comment why its doing that.

Reply:
Yeah, I don't know what's best. I could agree with any of them "most common" or "out of caution", but the comment is a good thing I think.